pkg/investigations/investigation.go

```diff
@@ -9,23 +9,35 @@ import (
 	hivev1 "github.com/openshift/hive/apis/hive/v1"
 )
 
+type InvestigationStep struct {
+	Performed bool
+	Labels    []string
+}
+
+type InvestigationResult struct {
+	LimitedSupportSet  InvestigationStep
+	ServiceLogPrepared InvestigationStep
+	ServiceLogSent     InvestigationStep
+}
+
 // Investigation serves as a parent class
 // This enables the structure of cmd/investigate.go
 type Investigation struct {
-	Run  func(resources *Resources) error
+	Run  func(resources *Resources) (InvestigationResult, error)
 	Name string
 }
 
 // NewInvestigation creates a new investigation
-func NewInvestigation(investigationFn func(resources *Resources) error, name string) *Investigation {
+func NewInvestigation(investigationFn func(resources *Resources) (InvestigationResult, error), name string) *Investigation {
 	return &Investigation{investigationFn, name}
 }
 
 // Resources holds all resources/tools required for alert investigations
 type Resources struct {
-	Cluster           *cmv1.Cluster
-	ClusterDeployment *hivev1.ClusterDeployment
-	AwsClient         aws.Client
-	OcmClient         ocm.Client
-	PdClient          pagerduty.Client
+	Cluster             *cmv1.Cluster
+	ClusterDeployment   *hivev1.ClusterDeployment
+	AwsClient           aws.Client
+	OcmClient           ocm.Client
+	PdClient            pagerduty.Client
+	AdditionalResources map[string]interface{}
```

Comment:
i think this can be generally nice to have ( this is added for the error to be passed to ccam? )

Are there other concerns with this besides remembering to do proper type assertion like you do above :D 
`bpError, ok := r.AdditionalResources["error"].(error)
if !ok {
	return result, fmt.Errorf("Missing required CCAM field 'error'")
}
`

Reply:
No that's the main concern, but I don't like it much - we have a statically typed language and this just kind of opts-out of it, but it would allow 'pass whatever you want' - or I can just add a dedicated error for just CCAM and keep it `nil` in all other cases.

Reply:
I see, 
I think both is completely fine to go forward with for now :) 

i wouldn't worry too much as we want to change this further most likely, 
if we have an alert which we can handle without the awsclient/cloud, we might want ccam to only add the LS and still remediate the resize or whatever?

Reply:
ah, this might get resolved when we properly chain the investigations 
the network verifier is called inside the chgm investigation for example instead of having its own investigation like ccam

but i feel like networkverifier and ccam are "basic checks" while chgm is alert -> basic check1, basic check2, nodes stopped? -> resolution/escalation

alert warrants investigation and always comes to a defined conclusion 
network verifier is informing but the chgm investigation decides what to do depending on network verifier output

these are non blocking comments, I am just thinking out loud haha^^